pkg/build/builder/docker.go

```diff
@@ -69,10 +70,19 @@ func (d *DockerBuilder) Build() error {
 	}
 	sourceInfo, err := fetchSource(d.dockerClient, buildDir, d.build, d.urlTimeout, os.Stdin, d.gitClient)
 	if err != nil {
+		d.build.Status.Reason = api.StatusReasonFetchSourceFailed
+		d.build.Status.Message = StatusMessageFetchSourceFailed
+		if updateErr := updateStatusRetrier(d.build, d.client); updateErr != nil {
```

Comment:
@bparees this was actually the first thing i've considered in the very begging, but i'll look again, maybe now with a better understanding it is possible.
what i remember was that the build process in the build pod is somewhat isolated from the controller. so i can't send updates from inside the buildpod to the controller that manages the build.
i'll still look into it, but... i don't know, maybe i'm missing something.